x/oracle/types/key_native_token.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import "strings"
+
+const (
+	// NativeTokenKeyPrefix is the prefix to retrieve all NativeToken
+	NativeTokenKeyPrefix                 = "NativeToken/"
+	NativeTokenPriceKeyPrefix            = NativeTokenKeyPrefix + "price/value/"
+	NativeTokenStakerInfoKeyPrefix       = NativeTokenKeyPrefix + "stakerInfo/value/"
+	NativeTokenOperatorInfoKeyPrefix     = NativeTokenKeyPrefix + "operatorInfo/value/"
+	NativeTokenStakerListKeyPrefix       = NativeTokenKeyPrefix + "stakerList/value/"
+	NativeTokenStakerDelegationKeyPrefix = NativeTokenKeyPrefix + "stakerDelegation/value/"
+
+// NativeTokenValidatorListKeyPrefix    = NativeTokenKeyPrefix + "validatorList/value"
+)
+
+func NativeTokenStakerDelegationKey(assetID, stakerAddr string) []byte {
+	assetID = strings.Join([]string{assetID, stakerAddr}, "/")
+	return append([]byte(NativeTokenStakerDelegationKeyPrefix), []byte(assetID)...)
+}
+
+func NativeTokenStakerListKey(assetID string) []byte {
+	return append([]byte(NativeTokenStakerListKeyPrefix), []byte(assetID)...)
+}
+
+func NativeTokenStakerKey(assetID, stakerAddr string) []byte {
+	assetID = strings.Join([]string{assetID, stakerAddr}, "/")
+	return append([]byte(NativeTokenStakerInfoKeyPrefix), []byte(assetID)...)
+}
+
+func NativeTokenStakerKeyPrefix(assetID string) []byte {
+	return append([]byte(NativeTokenStakerInfoKeyPrefix), []byte(assetID)...)
+}
+
+func NativeTokenOperatorKey(assetID, operatorAddr string) []byte {
+	assetID = strings.Join([]string{assetID, operatorAddr}, "/")
+	return append([]byte(NativeTokenOperatorInfoKeyPrefix), []byte(assetID)...)
+}
+
+// func NativeTokenValidatorListKey(assetID string) []byte {
+// 	return append([]byte(NativeTokenValidatorListKeyPrefix), []byte(assetID)...)
+// }
+
+func ParseNativeTokenStakerKey(key []byte) (assetID, stakerAddr string) {
+	parsed := strings.Split(string(key), "/")
```

Comment:
Use SplitN to limit the number of divisions

>parsed := **strings.SplitN**(string(key), "/", 4)

Reply:
SplitN will split string into slice with lenght=N and keep the remaining as a single string, that is not expectec.
We've got check `if len(parsed)!=3` ....